inputs/abeta_sims/part_1.py

```diff
@@ -0,0 +1,365 @@
+debug = False
+pickle_name = name + '.pickle'
+MPI_NP = 4
+peptide_copies = int(sys.argv[3])
+
+# try to reload
+if(os.path.exists(pickle_name)):
+    print('loading restart')
+    with open(pickle_name, 'rb') as f:
+        ps = pickle.load(f)
+else:
+    ps = PeptideSim(name, [seq], [peptide_copies], job_name='{}'.format(name))
+    ps.mdrun_driver = 'gmx_mpi'
+    ps.forcefield = 'amber99sb'
+    ps.water = 'tip4p'
+    ps.peptide_density = 0.008  # mg/ml
+    ps.ion_concentration = 0.001  # 10mM
+    ps.initialize()
+    with open(ps.pickle_name, 'wb') as f:
+        pickle.dump(ps, file=f)
+
+file00 = ps.pdb_file
+cwd = os.getcwd()
+# gromacs.pdb2gmx(f=file00,o="file_pdb_output.pdb",water=ps.water,ff=ps.forcefield)
+# file00="{}/{}".format(cwd,"file_pdb_output.pdb")
+# print(file00,"file_pdb_output.pdb", cwd)
+
+
+def total_aa(file1, output_file):
+    print(ps.sims[-1].location)
+    output = open(output_file, 'w')
+    with open(file1, "r") as f:
+        lines = f.readlines()
+        i = 4
+
+        for line in lines[:4]:
+            output.write(line)
+
+        for line in lines[4:]:
+            old_line = line
+            line = line.split()
+            i = i + 1
+            if(len(line) > 1 and line[3] == 'SOL'):
+
+                output.write('TER final\n')
+                output.write('ENDMDL final\n')
+                output.close()
+                break
+            # i=i+1
+            output.write(old_line)
+        last_line = []
+        print(lines[i - 1], lines[i - 2], lines[i - 3])
+        if (lines[i - 1].startswith('END')):
+
+            last_line = lines[i - 3].strip()
+            print("ends with END")
+        else:
+            last_line = lines[i - 1].strip()
+            print("doesnt end with END")
+
+        last_line = last_line.split()
+
+        output.close()
+        if (last_line[4].isdigit()):
+            return int(last_line[1]), int(last_line[4]), output_file
+        else:
+            return int(last_line[1]), int(last_line[5]), output_file
+# print total_aa(file00,'template.pdb')
+
+
+def number_all_atoms_chains():
+    if (os.path.isdir("{}/".format(os.getcwd) + "data")):
+        template_file = "{}/".format(os.getcwd) + "data/template.pdb"
+        # template_file="{}/".format(os.getcwd)+"data/template.pdb"
+        with open(template_file, 'r') as f:
+            lines = f.readlines()
+            lines = lines[-3].strip()
+            lines = lines.split()
+            return int(lines[1]), int(last_lines[5])
+
+
+total_no_atoms, number_chains, file0 = total_aa(file00, 'template.pdb')
+number_chains = int(number_chains)
+total_no_atoms = int(total_no_atoms)
+peptide_copies = int(peptide_copies)
+atoms_in_chain = int(total_no_atoms / peptide_copies)
+
+
+with open(file0, 'r') as file_read:
+    file_data = file_read.read()
+file_data = file_data.replace('HIS', 'HIE')
+with open(file0, 'w') as file_write:
+    file_write.write(file_data)
+
+
+def pdbfile_generator_w_chain_id(
```

Comment:
Just curious, does this function work for more than 2 copies yet?

Reply:
No I did not check with multiple copies